projects/Rockchip/bootloader/install

```diff
@@ -3,63 +3,53 @@
 
 PKG_RKBIN="$(get_build_dir rkbin)"
 PKG_SOC="${DEVICE/RK/rk}"
+UBOOT_CONFIG=$("${ROOT}"/"${SCRIPTS}"/uboot_helper "${PROJECT}" "${DEVICE}" "${UBOOT_SYSTEM}" config)
 
-case "${PKG_SOC}" in
-  rk3036)
-    PKG_DATAFILE="spl/u-boot-spl-nodtb.bin"
-    PKG_LOADER="u-boot-dtb.bin"
-    ;;
-  rk3288)
-    PKG_DATAFILE="${PKG_RKBIN}/rk32/rk3288_ddr_400MHz_v1.08.bin"
-    PKG_LOADER="${PKG_RKBIN}/rk32/rk3288_miniloader_v2.58.bin"
-    PKG_LOAD_ADDR="0x0"
-    ;;
-  rk3328)
-    PKG_DATAFILE="${PKG_RKBIN}/rk33/rk3328_ddr_786MHz_v1.16.bin"
-    PKG_LOADER="${PKG_RKBIN}/rk33/rk3328_miniloader_v2.50.bin"
-    PKG_BL31="${PKG_RKBIN}/rk33/rk3328_bl31_v1.42.elf"
-    PKG_BL31_ADDR="0x10000"
-    PKG_LOAD_ADDR="0x200000"
-    ;;
-  rk3399)
-    PKG_DATAFILE="${PKG_RKBIN}/rk33/rk3399_ddr_800MHz_v1.24.bin"
-    PKG_LOADER="${PKG_RKBIN}/rk33/rk3399_miniloader_v1.24.bin"
-    PKG_BL31="${PKG_RKBIN}/rk33/rk3399_bl31_v1.31.elf"
-    PKG_BL31_ADDR="0x40000"
-    PKG_LOAD_ADDR="0x200000"
-    ;;
-  *)
-    PKG_DATAFILE="spl/u-boot-spl-dtb.bin"
-    PKG_LOADER="u-boot-dtb.bin"
-    ;;
-esac
+if [ "${ROCKCHIP_LEGACY_BOOT}" = "1" ]; then
+  echo "install: building Rockchip legacy boot chain ..."
 
-if [ -n "${PKG_DATAFILE}" -a -n "${PKG_LOADER}" ]; then
-  tools/mkimage -n ${PKG_SOC} -T rksd -d "${PKG_DATAFILE}" idbloader.img
-  cat "${PKG_LOADER}" >> idbloader.img
-  cp -av idbloader.img ${INSTALL}/usr/share/bootloader
-fi
+  if [ ! -f u-boot-dtb.bin ]; then
+    echo "install: u-boot-dtb.bin does not exist UBOOT_TARGET: ${UBOOT_TARGET}"
+    exit 1
+  fi
+
+  PKG_BOOT_INI="${PKG_RKBIN}"/RKBOOT/"${DEVICE}"MINIALL.ini
+  if [ ! -f "${PKG_BOOT_INI}" ]; then
+    echo "install: ${PKG_BOOT_INI} does not exist"
+    exit 1
+  fi
+
+  PKG_DDR_BIN="${PKG_RKBIN}"/$(sed -n "/FlashData/s/FlashData=//p" "${PKG_BOOT_INI}")
+  PKG_MINILOADER="${PKG_RKBIN}"/$(sed -n "/FlashBoot/s/FlashBoot=//p" "${PKG_BOOT_INI}")
+  PKG_LOAD_ADDR=$(sed -n "/SYS_TEXT_BASE/s/#define CONFIG_SYS_TEXT_BASE //p" u-boot.cfg)
+
+  # Override sdram frequency
+  if [ "${DEVICE}" = "RK3328" ]; then
+    sed -s 's/\x4d\x1\x4d\x1\x4d\x1\x4d\x1\x4d\x1\x4d\x1/\x20\x3\x20\x3\x20\x3\x20\x3\x20\x3\x20\x3/g' -i "${PKG_DDR_BIN}"
+    sed -s 's/\x90\x1\x90\x1\x90\x1\x90\x1\x90\x1\x90\x1/\x20\x3\x20\x3\x20\x3\x20\x3\x20\x3\x20\x3/g' -i "${PKG_DDR_BIN}"
+  fi
```

Comment:
what this magic strings mean?

Reply:
As per comment: They replace dram-frequency in the binaries - in the first line  0x014d => 333 MHz with 800 MHz (0x0320).  Second line the same for 400 MHz. That's because RK does not offer binaries with higher frequencies. Previously we used to take dated ones but that no longer works for newer miniloader versions. 

You can completly ignore this if you're using an sbc which is supported by upstream u-boot. This is only used for legacy boot chain which is used by RK3328 TV boxes.

Reply:
thank you.